tools/misbehaving-jmx-server/src/main/java/org/datadog/supervisor/App.java

```diff
@@ -0,0 +1,159 @@
+package org.datadog.supervisor;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.concurrent.atomic.AtomicBoolean;
+import lombok.extern.slf4j.Slf4j;
+
+import com.mashape.unirest.http.HttpResponse;
+import com.mashape.unirest.http.Unirest;
+
+import io.javalin.*;
+
+@Slf4j
+class AppConfig {
+    public int rmiPort;
+    public int controlHttpPort;
+    public String rmiInterface;
+
+    public AppConfig() {
+        String rmiPortStr = System.getenv("JJ_RMI_PORT");
+        if (rmiPortStr != null) {
+            this.rmiPort = Integer.parseInt(rmiPortStr);
+        } else {
+            this.rmiPort = 1099;
+        }
+
+        String controlHttpPortStr = System.getenv("JJ_CONTROL_HTTP_PORT");
+        if (controlHttpPortStr != null) {
+            this.controlHttpPort = Integer.parseInt(controlHttpPortStr);
+        } else {
+            this.controlHttpPort = 8080;
+        }
+
+        this.rmiInterface = System.getenv("JJ_RMI_INTERFACE");
+        if (this.rmiInterface == null) {
+            this.rmiInterface = "localhost";
+        }
+    }
+}
+
+
+@Slf4j
+public class App {
+    private static Process process = null;
+    private static AtomicBoolean running = new AtomicBoolean(false);
+    private static final String jmxServerEntrypoint = "org.datadog.misbehavingjmxserver.App";
+    private static String selfJarPath;
+    private static final int jmxServerHTTPControlPort = 44751;
```

Comment:
this was chosen arbitrarily, would it be worthwhile to ask `misbehaving-jmx-server` to use an ephemeral port to bind to and then retrieve that port somehow?
Probably more effort than its worth for the small chance of a port confluct